feat: add exercise E The Odin Project (#50287)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/16-the-odin-project/top-learn-css-foundations-projects/css-foundations-exercise-e.md

# --description--

Understanding how combinators work can become a lot easier when you start playing around with them and see what exactly is affected by them versus what isn't.

The goal of this exercise is to apply styles to elements that are descendants of another element, while leaving elements that aren't descendants of that element unstyled.

> 1. You should see a `yellow` background, `p` elements that are descendants of the `div` element.
> 1. You should see a text color of `red` elements that are descendants of the `div` element.
> 1. You should see a font size of `20px` elements that are descendants of the `div` element.
> 1. You should text that is center aligned for elements that are descendants of the `div` element.

Reviewer: ```suggestion
1. You should see a `yellow` background for `p` elements that are descendants of the `div` element.
1. You should see a text color of `red` for elements that are descendants of the `div` element.
1. You should see a font size of `20px` for elements that are descendants of the `div` element.
1. You should center align text for elements that are descendants of the `div` element.
```

# --hints--

> You should have a background color of `yellow` on your decendants.

Reviewer: ```suggestion
You should have a background color of `yellow` on your descendants.
```

```js
const styleOne = new __helpers.CSSHelp(document).getStyle('.container .text');
const styleTwo =  new __helpers.CSSHelp(document).getStyle('div p');
const styleThree =  new __helpers.CSSHelp(document).getStyle('div .text');
const styleFour =  new __helpers.CSSHelp(document).getStyle('.container p');

const styles = [styleOne, styleTwo, styleThree, styleFour];

function getCorrectStyle(){
    for(let i = 0; i < styles.length; i++){
        if(styles[i] != undefined){
            return styles[i];
        }
    }
}

assert(getCorrectStyle()?.backgroundColor === 'yellow');

```

> You should have a text color of `red` on your decendants.

Reviewer: ```suggestion
You should have a text color of `red` on your descendants.
```

```js
const styleOne = new __helpers.CSSHelp(document).getStyle('.container .text');
const styleTwo =  new __helpers.CSSHelp(document).getStyle('div p');
const styleThree =  new __helpers.CSSHelp(document).getStyle('div .text');
const styleFour =  new __helpers.CSSHelp(document).getStyle('.container p');

const styles = [styleOne, styleTwo, styleThree, styleFour];

function getCorrectStyle(){
    for(let i = 0; i < styles.length; i++){
        if(styles[i] != undefined){
            return styles[i];
        }
    }
}

assert(getCorrectStyle()?.color === 'red');

```

> You should have a font size of `20px` on your decendants.

Reviewer: ```suggestion
You should have a font size of `20px` on your descendants.
```

```js
const styleOne = new __helpers.CSSHelp(document).getStyle('.container .text');
const styleTwo =  new __helpers.CSSHelp(document).getStyle('div p');
const styleThree =  new __helpers.CSSHelp(document).getStyle('div .text');
const styleFour =  new __helpers.CSSHelp(document).getStyle('.container p');

const styles = [styleOne, styleTwo, styleThree, styleFour];

function getCorrectStyle(){
    for(let i = 0; i < styles.length; i++){
        if(styles[i] != undefined){
            return styles[i];
        }
    }
}

assert(getCorrectStyle()?.fontSize === '20px');

```

> You should center align the text on your decendants.

Reviewer: ```suggestion
You should center align the text on your descendants.
```

# --description--

Understanding how combinators work can become a lot easier when you start playing around with them and see what exactly is affected by them versus what isn't.

The goal of this exercise is to apply styles to elements that are descendants of another element, while leaving elements that aren't descendants of that element unstyled.

1. You should see a `yellow` background, `p` elements that are descendants of the `div` element.
1. You should see a text color of `red` elements that are descendants of the `div` element.
1. You should see a font size of `20px` elements that are descendants of the `div` element.
1. You should text that is center aligned for elements that are descendants of the `div` element.

> 
> red text, a font size of 20px, and center aligned.

Reviewer 2: ```suggestion
```

const styleTwo =  new __helpers.CSSHelp(document).getStyle('div p');
const styleThree =  new __helpers.CSSHelp(document).getStyle('div .text');
const styleFour =  new __helpers.CSSHelp(document).getStyle('.container p');

const styles = [styleOne, styleTwo, styleThree, styleFour];

function getCorrectStyle(){
    for(let i = 0; i < styles.length; i++){
        if(styles[i] != undefined){
            return styles[i];
        }
    }
}

assert(getCorrectStyle()?.backgroundColor === 'yellow');

```

You should have a text color of `red` on your decendants.

```js
const styleOne = new __helpers.CSSHelp(document).getStyle('.container .text');
const styleTwo =  new __helpers.CSSHelp(document).getStyle('div p');
const styleThree =  new __helpers.CSSHelp(document).getStyle('div .text');
const styleFour =  new __helpers.CSSHelp(document).getStyle('.container p');

const styles = [styleOne, styleTwo, styleThree, styleFour];

function getCorrectStyle(){
    for(let i = 0; i < styles.length; i++){
        if(styles[i] != undefined){
            return styles[i];
        }
    }
}

assert(getCorrectStyle()?.color === 'red');

> 
> console.log(getCorrectStyle()?.color)

Reviewer 2: ```suggestion
```
